src/toil/wdl/wdltoil.py

```diff
@@ -1942,24 +1942,29 @@ def run(self, file_store: AbstractFileStore) -> Promised[WDLBindings]:
                 # their commas when separating the list, like in Cromwell's
                 # examples, so we strip whitespace.
                 spec_parts = spec.strip().split(' ')
+
+                # First check that this is a format we support. Both the WDL spec and Cromwell allow a max 3-piece specification
+                # So if there are more than 3 pieces, raise an error
+                if len(spec_parts) > 3:
+                    raise RuntimeError(f"Could not parse disks = {disks_spec} because {all_specs} contains more than 3 parts")
                 part_size = None
                 # default to GiB as per spec
-                part_suffix: str = "GiB"
+                part_suffix: str = "GiB"  # The WDL spec's default is 1 GiB
                 # default to the execution directory
                 specified_mount_point = None
+                # first get the size, since units should always be some nonnumerical string, get the last numerical value
                 for i, part in enumerate(spec_parts):
```

Comment:
The loop visits the parts in order, and with the `break` later `part_size` ends up being set to the *first* numerical value.